src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.DecisionDagRewriter.cs

```diff
@@ -137,13 +137,39 @@ public override BoundNode VisitCall(BoundCall node)
                     bool mightMutate =
                         // might be a call to a local function that assigns something
                         node.Method.MethodKind == MethodKind.LocalFunction ||
-                        // or perhaps we are passing a variable by ref and mutating it that way
-                        !node.ArgumentRefKindsOpt.IsDefault;
+                        // or perhaps we are passing a variable by ref and mutating it that way, e.g. `int.Parse(..., out x)`
+                        !node.ArgumentRefKindsOpt.IsDefault ||
+                        // or perhaps we are calling a mutating method of a value type
+                        MethodMayMutateReceiver(node.ReceiverOpt, node.Method);
```

Comment:
>node.ReceiverOpt [](start = 48, length = 16)

It doesn't look like ```MethodMayMutateReceiver``` is going to handle null as the first parameter.

Reply:
(see previous respondse)